src/store/modules/nft.js

```diff
@@ -30,7 +31,8 @@ const state = () => ({
 
 const mutations = {
   [TYPES.NFT_SET_NFT_CLASS_PURCHASE_INFO](state, { classId, info }) {
-    Vue.set(state.purchaseInfoByClassIdMap, classId, info);
+    const gemLevel = getGemLevelByCount(info?.metadata?.soldCount);
+    Vue.set(state.purchaseInfoByClassIdMap, classId, { gemLevel, ...info });
```

Comment:
I think should move the part `getGemLevelByCount(info?.metadata?.soldCount)` to getters `getNFTClassGemLevel`, If another mutation updates the purchase info, the gem level will be out sync